### PR TITLE
#817 make lint not running megacheck inside of travis

### LINKED ISSUE #817: make lint not running `megacheck` linter inside travis
# Problem

`gometalinter` is not running `megacheck` linter inside travis.

By running `make lint` locally you can force `megacheck` to break lint command,
```
$ time make lint
lint
geth/transactions/transactor.go:44:1:warning: unused is unused (deadcode)
geth/jail/doc.go:1:1:warning: package comment should be of the form "Package jail ..." (golint)
geth/transactions/transactor.go:44:6:warning: func unused is unused (U1000) (megacheck)
make: *** [lint] Error 1
make lint  88.81s user 34.33s system 310% cpu 39.653 total
```
however, when you run the same command inside travis the result is:
```
$ make lint
lint
geth/transactions/transactor.go:44:1:warning: unused is unused (deadcode)
make: *** [lint] Error 1
```

# Implementation

gometalinter is being installed on [Makefile](https://github.com/status-im/status-go/blob/develop/Makefile#L149) and it's being called from [.travis.yml](https://github.com/status-im/status-go/blob/develop/.travis.yml#L16).

# Acceptance Criteria

A `megacheck` linter error breaks the build on travis-ci

### PR DESCRIPTION
**Pull request by alexkohler**
After running gometalinter in debug mode, it was found megacheck was being killed by travis due to reaching its memory limits. For more information, see [this comment](https://github.com/status-im/status-go/issues/817#issuecomment-380848275).

@adriacidre was able to help in debugging with a couple useful suggestions:
* Attempt to increase memory via ulimit as suggested [here](https://github.com/status-im/status-go/issues/817#issuecomment-381061503) - I tried this with a couple different numbers, including `unlimit -s unlimied`
* There was also a suggestion to run a fully virtualized environment instead of a container-based one, which appears to fix the issue.

Here is a build that proves both megacheck and deadcode found the unused function: https://travis-ci.org/status-im/status-go/jobs/367728947 (from commit 5b25689)

Closes #817

### CODE REVIEW

**Comment by alexkohler** [comment]
@adriacidre - No luck here with `ulimit`, it appears a couple projects even use `ulimit -s unlimited`, which also doesn't seem to ameliorate megacheck's memory usage problem here

**Comment by adriacidre** [comment]
Hey @alexkohler i had some time to look into this. Even the [results are good enough](https://travis-ci.org/status-im/status-go/jobs/367334073#L535), the conclusions are not so clear. 

I've introduced [some changes on .travis.yml](https://github.com/status-im/status-go/compare/travis-mem-limits?expand=1#diff-354f30a63fb0907d4ad57269548329e3L14), so the stage `Lint & Vendor Check` is now running using a `fully virtualized environment` instead a container-based one.

Some docs about different [trusty environments](https://docs.travis-ci.com/user/reference/trusty/):
```
Fully-virtualized via sudo: required #

When specifying sudo: required, Travis CI runs each build in an isolated Google Compute Engine virtual machine that offer a vanilla build environment for every build.

This has the advantage that no state persists between builds, offering a clean slate and making sure that your tests run in an environment built from scratch.

Builds have access to a variety of services for data storage and messaging, and can install anything that’s required for them to run.

```

Please have a look at this solution, and feel free to use it on your pull request if everything looks good :)

**Comment by alexkohler** [comment]
@adriacidre Using a fully virtualized environment seems to do the trick - thank you! :)

**Comment by adriacidre** [comment]
awesome @alexkohler! please remember to rebase your branch ;)

**Review by adriacidre — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -12,6 +12,7 @@ install:
 jobs:
   include:
   - stage: Lint & Vendor Check
+    sudo: required
```
Comment (adambabik) [inline .travis.yml:15]: Any implication of this like longer startup time? If it's like 10-20 seconds it does not matter anyway.

**Review by mandrigin — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (alexkohler) [reply .travis.yml:15]: Looking at the recent Travis build times, it doesn't look like it: https://travis-ci.org/status-im/status-go/pull_requests In fact, my build took 21min 18s, which seems shorter than most of the builds around it

**Review by adambabik — APPROVED** [review]

**Pull request merged**